stagelib/envglobals.py

```diff
@@ -0,0 +1,19 @@
+import logging
```

Comment:
nothing in this file will ever work if one of these environment variables is undefined. The `IMAGES_TO_BUILD` will break in split if that is unset, the `pringGlobal` is not necessary because there is no need to ever check the `globals()` in Python to see if a variable is defined in a module that is coming from an environment variable. That function does not need to exist, it can only be logged or printed directly from looking at the environment variables.

Reply:
This is intended. The file explicitly should not work if any of the environment variables are undefined.

Reply:
There are many ways code can be made to stop executing if the desire is to not continue when a certain condition is not met. Defining environment variables in this particular way, inspecting `globals()` and calling `.split()` on something that might not work is not intuitive, does not help debugging, and will not explicitly say what was required or what was missing.

It will raise a KeyError on the first item. What if you are missing 3 out of 4, or more than one? It will not tell you.

Reply:
That can be improved.

Reply:
I agree that showing a traceback is rarely a good option and that's why I suggested using explicit error handling to clarify that.
About the variables, they come from the Makefile (which is the only way to use it) which populates them.

Reply:
I have added exception handling to the `envglobals.py` module to fix this issue.

Reply:
I don't understand why envglobals.py exists either; also see prior comments about printGlobal just being wrong